code/mob/melee_attack_procs.dm

```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
+
+		if (P?.barbed == FALSE)
+			src.visible_message("<span class='alert'><B>[src] pulls a dart out of [target]!</B></span>")
+			P.on_remove(M)
+			M.implant.Remove(P)
+			P.set_loc(M.loc)
+		else if (P?.barbed == TRUE)
+			src.visible_message("<span class='alert'><B>[src] tries to pull a dart out of [target], but it's stuck in!</B></span>")
+
+		if (!P || P?.barbed == TRUE)
+			P = locate(/obj/item/implant/projectile/syringe) in M.implant
+
+			if (P?.barbed == FALSE)
+				src.visible_message("<span class='alert'><B>[src] pulls a syringe out of [target]!</B></span>")
+				P.on_remove(M)
+				M.implant.Remove(P)
+				P.set_loc(M.loc)
+			else if (P?.barbed == TRUE)
+				src.visible_message("<span class='alert'><B>[src] tries to pull a syringe out of [target], but it's stuck in!</B></span>")
+
```

Comment:
Would like to return after successfully pulling an implant out; feels weird brushing off your shoes every time you pull a dart out of yourself.